Extend/Controller/OrderController.php

```diff
@@ -273,10 +369,126 @@ private function _isPaymentFinished($oSessionToken, $oPaymentRedirect)
+
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+
+        /**
+         * @var $oOrder Order
+         */
+        $oOrder = oxNew(Order::class);
+        $oOrder->createTemp($oBasket, $this->getUser());
+
+        $oTransaction = $oPaymentGateway->createTransaction($oBasket, $oOrder);
+
+        Registry::getLogger()->debug("getInitCreditCardFormJavaScript Transaction:" . print_r($oTransaction, true));
+
+        $oTransaction->setAmount(new Amount(
+            $oBasket->getPrice()->getBruttoPrice(),
+            $oBasket->getBasketCurrency()->name
+        ));
+        $oTransaction->setConfig($this->_getPaymentMethodConfig()->get(CreditCardTransaction::NAME));
+
+        $oSession = $this->getSession();
+        $sSid = $oSession->sid(true);
+        if ($sSid != '') {
+            $sSid = '&' . $sSid;
+        }
+
+        $sModuleToken = PaymentGateway::getModuleToken($oSession);
+        $oTransaction->setTermUrl(
+            $this->getConfig()->getCurrentShopUrl() . "index.php?cl=order" . $sModuleToken . $sSid
+        );
+
+        $oTransactionService = $this->_getTransactionService();
+
+        $oPayment = oxNew(Payment::class);
+        $oPayment->load($oBasket->getPaymentId());
+
+        return "CreditCardForm.init(" .
+            $oTransactionService->getCreditCardUiWithData(
+                $oTransaction,
+                $this->_getPaymentAction($oPayment->oxpayments__wdoxidee_transactionaction->value),
+                Registry::getLang()->getLanguageAbbr()
+            ) . ")";
+    }
+
+    /**
+     * @return TransactionService
+     * @throws \Exception
+     */
+    private function _getTransactionService(): TransactionService
+    {
+        if (is_null($this->_oTransactionService)) {
+            $this->_oTransactionService = new TransactionService($this->_getPaymentMethodConfig());
+        }
+
+        return $this->_oTransactionService;
+    }
+
+    /**
+     * @return CreditCardPaymentMethod
+     * @throws \Exception
+     */
+    private function _getCreditCardPaymentMethod(): CreditCardPaymentMethod
+    {
+        if (is_null($this->_oCreditCardPaymentMethod)) {
+            $this->_oCreditCardPaymentMethod = new CreditCardPaymentMethod();
+        }
+
+        return $this->_oCreditCardPaymentMethod;
+    }
+
+    /**
+     * @return Config
+     * @throws \Exception
+     */
+    private function _getPaymentMethodConfig(): Config
+    {
+        if (is_null($this->_oPaymentMethodConfig)) {
+            $oPayment = $this->getPayment();
+            $this->_oPaymentMethodConfig = $this->_getCreditCardPaymentMethod()->getConfig($oPayment);
+        }
+
+        return $this->_oPaymentMethodConfig;
+    }
+
+    /**
+     * Converts the admin panel payment method action into a seamless
+     *
+     * @param string $sAction
+     *
+     * @return string
+     */
+    private function _getPaymentAction(string $sAction): string
```

Comment:
Is this specific to credit card? If not, we might want to put it somewhere else.

Reply:
Actually, it is specific to the Javascript form provided by the payment SDK. So every other Payment method with that behavior will use it. I think we will not need it anywhere else.